conda_forge_tick/new_update_versions.py

```diff
@@ -0,0 +1,148 @@
+    gx: nx.DiGraph, sources: Iterable[AbstractSource] = None
+) -> None:
+    sources = (
+        (PyPI(), CRAN(), NPM(), ROSDistro(), RawURL(), Github())
+        if sources is None
+        else sources
+    )
+
+    _all_nodes = [t for t in gx.nodes.items()]
+    random.shuffle(_all_nodes)
+
+    # Inspection the graph object and node update:
+    # print(f"Number of nodes: {len(gx.nodes)}")
+    Node_count = 0
+
+    to_update = []
+    for node, node_attrs in tqdm.tqdm(_all_nodes):
+        attrs = node_attrs["payload"]
+        if attrs.get("bad") or attrs.get("archived"):
+            attrs["new_version"] = False
+            continue
+        to_update.append((node, attrs))
+
+    up_to = {}
+    for node, node_attrs in to_update:
+        # checking each node
+        with node_attrs as attrs:
+            up_to = {}
+
+            # verify the actual situation of the package;
+            actual_ver = str(attrs.get("version"))
+
+            # rude exception
+            if node == "ca-policy-lcg":
+                up_to["ca-policy-lcg"] = {
+                    "bad": attrs.get("bad"),
+                    "new_version": False,
+                    "new_version_attempts": attrs.get("new_version_attempts"),
+                    "new_version_errors": attrs.get("new_version_errors")
+                }
+                Node_count += 1
+                continue
+
+            # New verison request
+            try:
+                new_version = get_latest_version(node, attrs, sources)
+                attrs["new_version"] = new_version or attrs["new_version"]
+            except Exception as e:
+                try:
+                    se = repr(e)
+                except Exception as ee:
+                    se = "Bad exception string: {}".format(ee)
+                logger.warning(
+                    f"Warning: Error getting upstream version of {node}: {se}"
+                )
+                attrs["bad"] = "Upstream: Error getting upstream version"
+            else:
+                logger.info(
+                    f"# {Node_count:<5} - {node} - {attrs.get('version')} - {attrs.get('new_version')}",
+                )
+            up_to[f"{node}"] = {
+                "bad": attrs.get("bad"),
+                "new_version": attrs.get("new_version"),
+                "new_version_attempts": attrs.get("new_version_attempts"),
+                "new_version_errors": attrs.get("new_version_errors")
+            }
+            Node_count += 1
+    return up_to
+
+
+def main(args: Any = None) -> None:
+    if CONDA_FORGE_TICK_DEBUG:
+        setup_logger(logger, level="debug")
+    else:
+        setup_logger(logger)
+
+    logger.info("Reading graph")
+    # Graph enabled for inspection
+    gx = load_graph()
+
+    # call update
+    to_update = new_update_upstream_versions(gx)
+
+    logger.info("writing out file")
+    with open("new_version.json", "w") as outfile:
+        json.dump(to_update, outfile)
```

Comment:
We should dump one file per feedstock.

Reply:
Is there a specific path we should use for them ? (like bump on it's Node path)

Reply:
Can I create a file on `cf-graph-countyfair` to save these files according to node ?(`cf-graph-countyfairversions/{node}_new_version.json`)

Reply:
You can make a folder using `os.makedirs(foldername, exist_ok=True)`. The files should just be named `{node}.json` since the folder name should be descriptive that they are version numbers.

Reply:
oh I see, them as we already have a Job on circle that saves all the alterations made during the running, them it will push the alterations automatically right ? If that's so I will made the corrections right away thanks :+1:

Reply:
Done, could someone check if it's ok ?